Userland/Libraries/LibCore/System.cpp

```diff
@@ -205,4 +205,27 @@ ErrorOr<String> gethostname()
     return String(&hostname[0]);
 }
 
+ErrorOr<void> chown(StringView pathname, uid_t uid, gid_t gid)
+{
+    int rc = ::chown(pathname.characters_without_null_termination(), uid, gid);
```

Comment:
This is a bug waiting to happen... You have to ensure that these are null terminated, see implementations above that use the raw syscall so they can pass the explicit length instead.

Same goes with `chmod` below.

Reply:
Fixed by using raw syscall for `chown()` and `chmod()`.